### PR TITLE
docs: Expand and fix documentation for jx gitops git subcommands

### PR DESCRIPTION
**Pull request by thecooldrop**
# Description

I have added more detailed descriptions of what the commands under `jx gitops git` do when they are executed. This should help the users to be more certain that they are using the commands correctly.

Fixes # (issue)

# Checklist:

- [x] I have mentioned the appropriate type(scope), as referenced [here](https://jenkins-x.io/community/code/#the-commit-message) in the commit message and PR title for the semantic checks to pass.
- [x] I have signed off the commit, as per instructions mentioned [here](https://jenkins-x.io/community/code/#how-to-sign-your-commits).
- [x] Any dependent changes have already been merged.

### CODE REVIEW

File: content/en/v3/develop/reference/jx/gitops/git/merge/_index.md [hunk 1]
```diff
@@ -9,23 +9,38 @@ aliases:
...
+
+This command merges a list of commits into a specified branch. If the branch does not exist among local branches, then
+it is first created.
+
+If both `--pull-refs` and `--sha` flags are specified then only those commits specified by `--sha` are merged into the
+base branch.
+
+If `--include-comment` or `--exclude-comment` flags are specified, then `--pull-number` flag needs to be set as well.
+If only one of `--include-comment` or `--exclude-comment`, then only that one is used to filter commits while other is
+ignored. If both are specified, then only those commits which satisfy ``--include-comment` and do not satisfy the
```
Comment (hervelemeur) [inline content/en/v3/develop/reference/jx/gitops/git/merge/_index.md:30]:
```suggestion
ignored. If both are specified, then only those commits which satisfy `--include-comment` and do not satisfy the
```

File: content/en/v3/develop/reference/jx/gitops/git/setup/_index.md [hunk 2]
```diff
@@ -25,14 +25,50 @@ This is typically used in a pipeline to ensure git can do commits.
   jx-gitops git setup
 
   ```
+  
+### Details
+
+The `jx gitops git setup` command ensures that we can authenticate with configured Git server by configuring the local
+credentials file in the home directory. This command tries to ensure following things:
```
Comment (hervelemeur) [inline content/en/v3/develop/reference/jx/gitops/git/setup/_index.md:32]:
```suggestion
credentials file in the home directory. This command tries to ensure the following things:
```

File: content/en/v3/develop/reference/jx/gitops/git/setup/_index.md [hunk 3]
```diff
@@ -25,14 +25,50 @@ This is typically used in a pipeline to ensure git can do commits.
...
+- value stored in `HOME` environment variable or
+- value stored in `USERPROFILE` environment variable or
+- as current directory `.`
+
+The credentials are determined by reading out the `jx-requirements.yaml` from the cluster repository and `jx-boot`
+Secret resource provisioned together with `jx-git-operator` in your Kubernetes namespace.
+
+The Git username and email are preferentially determined from `PipelineUser` field from `jx-requirements.yaml`, but if
+they are not available there then default email address `jenkins-x@googlegroups.com` is used (which is the reason why
+you are seeing commits from `pow-devops2020` user made on your repositories). If the username could not be determined
```
Comment (hervelemeur) [inline content/en/v3/develop/reference/jx/gitops/git/setup/_index.md:49]: Shouldn't be the case anymore for new users, not sure if pow-devops2020 should be mentioned.

**Review by hervelemeur — CHANGES_REQUESTED** [review]
Great!

File: content/en/v3/develop/reference/jx/gitops/git/setup/_index.md (hunk 3 above)
Comment (thecooldrop) [reply content/en/v3/develop/reference/jx/gitops/git/setup/_index.md:49]: Does this mean that in newer Jx versions email is going to be mandatory part of git access configuration? If yes then I agree.

Comment (hervelemeur) [reply content/en/v3/develop/reference/jx/gitops/git/setup/_index.md:49]: I don't know, but it would make sense IMO.

Comment (thecooldrop) [reply content/en/v3/develop/reference/jx/gitops/git/setup/_index.md:49]: @jstrachan has commented on this topic in Slack conversation, that something was in the making about usage of `pow-devops2020` user. Maybe he can provide more info?

**Comment by jstrachan** [comment]
@thecolddrop thanks for the PR. Unfortunately all the docs you are modifying are code generated from the underlying source code.

So the code is in jx-gitops: https://github.com/jenkins-x-plugins/jx-gitops/tree/main/docs/cmd

which is actually code generated from the documentation inside the actual go code for the command. e.g. for the git clone command its here: https://github.com/jenkins-x-plugins/jx-gitops/blob/main/pkg/cmd/git/clone/clone.go#L24-L53 which is then used to generate all the markdown/CLI help/website docs

**Comment by thecooldrop** [comment]
I have moved these contributions into the source code as advised. Please review the [pull request on jx-gitops repo PR](https://github.com/jenkins-x-plugins/jx-gitops/pull/766). @hervelemeur

**Pull request closed without merging**